packages/components/src/List/ListComposition/Manager/hooks/useCollectionSelection.hook.js

```diff
@@ -7,20 +7,20 @@ export default function useCollectionSelection(
 ) {
 	const [selectedIds, setSelectedIds] = useState(initialSelectedIds);
 
-	function filterSelectionFromCollection(selection) {
-		return collection
-			.filter(item => selection.includes(item[idKey]))
-			.map(item => item[idKey]);
-	}
+	const filterSelectionFromCollection = useCallback(
+		selection =>
+			collection.filter(item => selection.includes(item[idKey])).map(item => item[idKey]),
+		[idKey, collection],
+	);
 
 	useEffect(() => {
-		if (selectedIds.length === 0) {
-			return;
-		}
-
-		const filteredSelection = filterSelectionFromCollection(selectedIds);
-		setSelectedIds(filteredSelection);
-	}, [collection]);
+		setSelectedIds(oldIds => {
+			if (oldIds.length === 0) {
+				return oldIds;
+			}
+			return filterSelectionFromCollection(oldIds);
+		});
+	}, [collection, filterSelectionFromCollection]);
```

Comment:
The issue here is that the useEffect uses 
- `collection` (in the hook dependency)
- `selectedIds/setSelectedIds` (not in deps)
- `filterSelectionFromCollection` (not in deps)

But 
- `filterSelectionFromCollection` is always a new ref at each render
- `selectedIds` can't be a dependency because we do a `setSelectedIds()` inside. It would introduce an infinite loop

So what has changed ?
- the useEffect only do the `setSelectedIds`
- removed the dependency to `selectedIds` by using a state callback to get the current `selectedIds`
- `filterSelectionFromCollection` is a dependency, but it's a `useCallback()` to have the same ref if its dependencies have not changed